### ISSUE TITLE
Typescriptifying the project

### ISSUE THREAD

**Issue by yigitozkavci** [yigitozkavci/wowwizard#4]
In `TypeScriptify` branch, I am porting whole project into `TypeScript`, and I need help. Any strict typers to lend me a hand?

**Comment by emanuelmd** [comment]
I could give this a try. Do you need a build system in place or just port the js files to typescript?

**Comment by yigitozkavci** [comment]
No, actually `gulp` handles that, in `TypeScriptify` branch, I created some of the classes, and project successfully runs on TypeScript. It just needs some more work, can you work on `TypeScriptify` branch @emanuelmd?

**Comment by emanuelmd** [comment]
Sure

**Comment by emanuelmd** [comment]
Sorry for the late reply, I managed to fix your typescript errors, but I suggest refactoring the whole project to take full advantage of Typescript. I could help you with that.

#20

**Comment by emanuelmd** [comment]
Also, the dev/ folder is a bit messy and the build system kind of unintuitive.

**Comment by yigitozkavci** [comment]
Yeap, that is possible since that branch wasn't meant to be edited by someone else yet, but now I am happy with the situation. https://github.com/yigitozkavci/wowwizard/pull/20 seems good to me, if you have any questions about the code, I would happily reply here or on PR page, thanks! @emanuelmd

**Comment by emanuelmd** [comment]
I'm having a few ideeas about the whole refactoring, I could commit on my fork and Pull Request into your Typescriptify branch when I have something. Alternatively I can present you my ideeas on some sort of chat service.

**Comment by yigitozkavci** [comment]
Pushing to that branch would be nice, since I also get to see those commits from #20. We can use Slack for communicating, how about a wowwizard Slack group?

**Comment by emanuelmd** [comment]
Sounds good, send me an email with the link or something.
